openfast_python/openfast_io/FAST_linearization_reader.py

```diff
@@ -0,0 +1,391 @@
+        return 'FAST linearization output'
+
+    def __init__(self, filename=None, **kwargs):
+        """ Class constructor. If a `filename` is given, the file is read. """
+        self.filename = filename
+        if filename:
+            self.read(**kwargs)
+
+    def read(self, filename=None, **kwargs):
+        """ Reads the file self.filename, or `filename` if provided """
+        
+        # --- Standard tests and exceptions (generic code)
+        if filename:
+            self.filename = filename
+        if not self.filename:
+            raise Exception('No filename provided')
+        if not os.path.isfile(self.filename):
+            raise OSError(2,'File not found:',self.filename)
+        if os.stat(self.filename).st_size == 0:
+            raise EmptyFileError('File is empty:',self.filename)
+        # --- Calling (children) function to read
+        self._read(**kwargs)
+
+    def _read(self, *args, **kwargs):
+        self['header']=[]
+
+        def extractVal(lines, key):
+            for l in lines:
+                if l.find(key)>=0:
+                    return l.split(key)[1].split()[0]
+            return None
+
+        def readToMarker(fid, marker, nMax):
+            lines=[]
+            for i, line in enumerate(fid):
+                if i>nMax:
+                    raise BrokenFormatError('`{}` not found in file'.format(marker))
+                if line.find(marker)>=0:
+                    break
+                lines.append(line.strip())
+            return lines, line
+        
+        def readOP(fid, n, name=''):
+            OP=[]
+            Var = {'RotatingFrame': [], 'DerivativeOrder': [], 'Description': []}
+            colNames=fid.readline().strip()
+            fid.readline().strip()
+            bHasDeriv= colNames.find('Derivative Order')>=0
+            for i, line in enumerate(fid):
+                sp = line.strip().split()
+                if sp[1].find(',') >= 0:
+                    #  Most likely this OP has three values (e.g. orientation angles)
+                    # For now we discard the two other values
+                    OP.append(np.float32(sp[1][:-1]))
+                    iRot=4
+                else:
+                    OP.append(np.float32(sp[1]))
+                    iRot=2
+                Var['RotatingFrame'].append(sp[iRot])
+                if bHasDeriv:
+                    Var['DerivativeOrder'].append(int(sp[iRot+1]))
+                    Var['Description'].append(' '.join(sp[iRot+2:]).strip())
+                else:
+                    Var['DerivativeOrder'].append(-1)
+                    Var['Description'].append(' '.join(sp[iRot+1:]).strip())
+                if i>=n-1:
+                    break
+            OP=np.asarray(OP)
+            return OP, Var
+
+        def readMat(fid, n, m, name=''):
+
+            # Read rows from file, raise exception on failure
+            try:
+                vals = np.genfromtxt(fid, dtype=np.float64, max_rows=n)
```

Comment:
@mayankchetan This doesn't necessarily need to be fixed here, but it turns out that genfromtxt uses a bunch of memory and it's more efficient to do  the following (https://stackoverflow.com/questions/49076386/why-does-np-genfromtxt-initially-use-up-a-large-amount-of-memory-for-large-dat) though you may want to check this code first:

```suggestion
                vals = np.empty([n,m], np.float64)
                for i in range(n):
                    vals[i,:] = fid.readline().split()
```

Reply:
Thank you for the heads up! will make the changes.